src/histogram.c

```diff
@@ -29,12 +38,226 @@ TS_FUNCTION_INFO_V1(hist_serializefunc);
 TS_FUNCTION_INFO_V1(hist_deserializefunc);
 TS_FUNCTION_INFO_V1(hist_finalfunc);
 
-/* histogram(state, val, min, max, nbuckets) */
+typedef struct SubAggState
```

Comment:
Can we document the roles of each of these structs? It is a bit hard to follow right now what each of them are used for.

Reply:
Done